src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -388,9 +388,22 @@ void MainWindow::build_menus()
     m_ui->menu_view->addSeparator();
 
     QAction* fullscreen_action = m_ui->menu_view->addAction("Fullscreen");
+    fullscreen_action->setCheckable(true);
     fullscreen_action->setShortcut(Qt::Key_F11);
+    fullscreen_action->setObjectName("Fullscreen");
+    connect(m_ui->project_explorer->toggleViewAction(), SIGNAL(triggered()), SLOT(slot_check_fullscreen()));
+    connect(m_ui->attribute_editor->toggleViewAction(), SIGNAL(triggered()), SLOT(slot_check_fullscreen()));
+    connect(m_ui->log->toggleViewAction(), SIGNAL(triggered()), SLOT(slot_check_fullscreen()));
+    connect(m_ui->python_console->toggleViewAction(), SIGNAL(triggered()), SLOT(slot_check_fullscreen()));
     connect(fullscreen_action, SIGNAL(triggered()), SLOT(slot_fullscreen()));
 
+    if (any_of(m_minimize_buttons.cbegin(),
```

Comment:
Simply call `slot_check_fullscreen()` to avoid logic duplication.

Reply:
This actually doesn't work anymore. I'm not sure what exactly changed between Qt4 and Qt5, but now the checkboxes load after everything else. I tried calling `slot_check_fullscreen` elsewhere in the code, but it is always setting the screen to fullscreen because the other checkboxes haven't loaded. Is there somewhere you recommend me look?

Reply:
Do I understand correctly that as it stands now (with the two new commits you made recently), it doesn't work?

Reply:
Everything works except during startup. I believe something changed in the Qt4 to Qt5 transition. I had programmed it so that the Fullscreen checkbox would be checked in the following scenario:

1. User unchecks all checkboxes
2. User closes appleseed.studio
3. User reopens appleseed.studio and finds that all checkboxes are unchecked except for the Fullscreen checkbox

However, now the Fullscreen checkbox is not checked in this case.

Reply:
Isn't the problem simply that you check the status of the checkboxes _before_ `slot_load_application_settings()` is called? It's called in `MainWindow`'s constructor.

Reply:
No, initially in my tests I called `slot_check_fullscreen()` after `slot_load_application_settings()` in the constructor, however this came with its own problems.

The checkboxes aren't checked when the MainWindow is constructed, thus the Fullscreen checkbox always starts as checked. To debug this I tried calling `slot_check_fullscreen()` after startup and everything worked as expected. This leads me to believe this problem has something to do with how things are initialized at startup.

Reply:
@dictoon I have fixed the issue in the latest commit. The problem was that I was calling the MinimizeButton `is_on()` method, but the method was not in sync with the QDockWidget `isHidden()` method. 

Now `findChildren() is being used to find all of the QDockWidgets. I'm not sure this should be an instance variable like m_action_fullscreen, but if you want me to make it one I will do so.